### PR TITLE
Upgrade php-cs-fixer to 3.0 & infection to 0.22.1

### PR DESCRIPTION
**Pull request by jmsche**
| Q                | A
| ---------------- | ---
| PHP version?     | N/A
| Bundle version?  | N/A
| Symfony version? | N/A
| New feature?     | no
| Bug fix?         | no
| Discussion?      | N/A

Upgrade php-cs-fixer to 3.0 & infection to 0.22.1.

Initially I wanted to upgrade php-cs-fixer only, but it requires composer/xdebug-handler^2, and only infection^0.22.1 supports it.

@Guikingone May I ask why some files other than daily.yaml have these lines? https://github.com/Guikingone/SchedulerBundle/blob/0.4.2/.github/workflows/code-style.yml#L9-L10
Seems overkill

### CODE REVIEW

**Comment by jmsche** [comment]
CI failures unrelated

File: composer.json [hunk 1]
```diff
@@ -101,8 +101,8 @@
         "api-platform/core": "^2.6",
         "doctrine/dbal": "^2.12 || ^3.0",
         "doctrine/orm": "^2.8",
-        "friendsofphp/php-cs-fixer": "^2.18",
-        "infection/infection": "^0.21",
+        "friendsofphp/php-cs-fixer": "^3.0.0",
+        "infection/infection": "^0.22.1",
```
Comment (Guikingone) [inline composer.json:105]: Could we use `^0.22` instead?

Comment (jmsche) [reply composer.json:105]: No: as said, 0.22.1 is the only version allowing xdebug-handler v2: https://packagist.org/packages/infection/infection#0.22.0

Comment (Guikingone) [reply composer.json:105]: Oh, sorry, didn't read the first comment, looks good to me, thanks 🙂

**Pull request merged**